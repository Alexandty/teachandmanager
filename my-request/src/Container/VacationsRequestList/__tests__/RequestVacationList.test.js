import React from 'react';
import { configure, shallow, render } from 'enzyme';
import { RequestVacationList } from '../index';
import ConnectedRequestVacationList from '../index';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';

const mockStore = configureStore()

configure({ adapter: new Adapter() });

describe('test Vacationlist', () => {
    it('renders  Vacation when store is empty', () => {
        const wrapper = shallow(<RequestVacationList loadRequestVacation={jest.fn()} VacationData={[{ idRequest: 1 }]} user={[{ 'name': 's' }]} />);
        expect(wrapper.find(".Vacation").length).toBe(1);
    });

    it('No debe mostrar Vacation cuando store este vacio', () => {
        const wrapper = shallow(<RequestVacationList loadRequestVacation={jest.fn()} VacationData={[]} user={{ name: 'juan' }} />);
        expect(wrapper.find(".ListRequestVacationEmpty").length).toBe(1);
    })

    it('Validando toMatchSnapshot RequestVacationList', () => {
        const props = {
            loadRequestVacation: jest.fn(),
            VacationData: [],
            user: { user: 'a', entryDate: '2018-08-09' }
        };
        const wrap = shallow(<RequestVacationList {...props} />);
        expect(toJson(wrap)).toMatchSnapshot();
    });

    it('renders and sort vacations', () => {
        const wrapper = shallow(<RequestVacationList loadRequestVacation={jest.fn()} VacationData={[{ idRequest: 1 }, { idRequest: 2 }]} user={[{ 'name': 's' }]} />);
        expect(wrapper.find(".Vacation").exists()).toBe(true);
    });

    it('renders the connected component', () => {
        const initialState = {}
        const store = mockStore(initialState)
        const wrapper = shallow(<ConnectedRequestVacationList store={store} loadRequestVacation={jest.fn()} VacationData={[{ idRequest: 1 }]} user={[{ name: 's' }]} />);
    });
})