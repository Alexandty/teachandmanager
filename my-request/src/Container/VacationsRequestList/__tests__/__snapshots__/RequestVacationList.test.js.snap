// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Vacationlist Validando toMatchSnapshot RequestVacationList 1`] = `
<div
  className="ListRequestVacationEmpty"
>
   
   
  <Label
    bsClass="label"
    bsStyle="success"
  >
    fecha de ingreso a la empresa 
    09/08/2018
  </Label>
  <Alert
    bsClass="alert"
    bsStyle="info"
    closeLabel="Close alert"
  >
    <h4>
      !Lo sentimos!
    </h4>
    <p>
      Usted no cuenta con informacion de solicitudes.
    </p>
  </Alert>
</div>
`;
